docs/requirements/database-structure-for-plan-entities.md

```diff
@@ -0,0 +1,149 @@
+| `is_completed`   | BOOLEAN                      | Whether the task is completed                                | **NOT NULL**                                            | FALSE              |
+| `execution_time` | execution\_time\_type (ENUM) | Execution time category (`morning`, `afternoon`, `evening`)  | **NOT NULL**                                            | 'morning'               |
+| `updated_at`     | TIMESTAMPTZ                  | Timestamp of last update                                     | **NOT NULL**                                            | CURRENT\_TIMESTAMP |
+| `completed_at`   | TIMESTAMPTZ                  | Timestamp when the task was completed                        | Nullable                                                | NULL               |
+| `is_custom`      | BOOLEAN                      | Whether task is created by user (`TRUE`) or system (`FALSE`) | **NOT NULL**                                            | FALSE              |
+| `parent_task_id` | INT                          | Optional reference to the task this was regenerated from     | Nullable, FK → `task(id)` **ON DELETE SET NULL**        | NULL               |
+
+# Constraints Explanation:
+
+- deleting a plan day deletes all its tasks (ON DELETE CASCADE).
+- if the referenced execution time is deleted, the execution_time_id in task is set to NULL.
+- parent_task_id is nullable and set to NULL if the referenced task is deleted.
+
+# 4. Table: task_translation
+
+Description: stores translations of tasks for multiple languages. Each task can have a title, description, and tip in different languages.
+
+| Column          | Type         | Description                                             | Constraints                                         | Default        |
+| --------------- | ------------ | ------------------------------------------------------- | --------------------------------------------------- | -------------- |
+| `id`            | SERIAL       | Unique identifier for each translation record           | PK                                                  | Auto-increment |
+| `task_id`       | INT          | Reference to the original task being translated         | **NOT NULL**, FK → `task(id)` **ON DELETE CASCADE** | None           |
+| `language_code` | VARCHAR(5)   | ISO language code for the translation (e.g. `en`, `es`) | **NOT NULL**, FK → `language(code)`                 | None           |
+| `title`         | VARCHAR(200) | Translated task title                                   | **NOT NULL**                                        | None           |
+| `description`   | TEXT         | Translated description for the task                     | Nullable                                            | NULL           |
+| `tip`           | TEXT         | Optional translated tip or hint for the task            | Nullable                                            | NULL           |
```

Comment:
I think it can be removed because we are going to have only English description. If we have time we will use localization on the frontend